addons/mail/models/mail_template.py

```diff
@@ -492,6 +492,8 @@ def _generate_template(self, res_ids, render_fields,
                 )
                 for res_id, field_value in generated_field_values.items():
                     render_results.setdefault(res_id, {})[field] = field_value
+                    if field == 'body_html':
```

Comment:
That's not the right place to do it. Template generates values based on asked fields, the caller has to use it accordingly. Template should not be aware of external name change. We cleaned that in 16.2 and we should not get that kind of strange behavior back :) .

And it definitively requires unit test(s).